ctapipe/ml/tests/test_sklearn.py

```diff
@@ -0,0 +1,139 @@
+import pytest
+
+import numpy as np
+from traitlets.config import Config
+from traitlets import TraitError
+from astropy.table import Table
+from numpy.testing import assert_array_equal
+
+
+def test_supported_regressors():
+    from ctapipe.ml.sklearn import SUPPORTED_REGRESSORS
+    from sklearn.ensemble import RandomForestRegressor
+
+    assert "RandomForestRegressor" in SUPPORTED_REGRESSORS
+    assert SUPPORTED_REGRESSORS["RandomForestRegressor"] is RandomForestRegressor
+
+
+def test_supported_classifiers():
+    from ctapipe.ml.sklearn import SUPPORTED_CLASSIFIERS
+    from sklearn.ensemble import RandomForestClassifier
+
+    assert "RandomForestClassifier" in SUPPORTED_CLASSIFIERS
+    assert SUPPORTED_CLASSIFIERS["RandomForestClassifier"] is RandomForestClassifier
+
+
+@pytest.fixture()
+def example_table():
+    from sklearn.datasets import make_regression
+    from sklearn.datasets import make_blobs
+
+    X, y = make_regression(n_samples=100, n_features=5, n_informative=3, random_state=0)
+    t = Table({f"X{i}": col for i, col in enumerate(X.T)})
+    t["energy"] = y
+    t["X0"][10] = np.nan
+    t["X1"][30] = np.nan
+
+    X, y = make_blobs(n_samples=100, n_features=3, centers=2, random_state=0)
+    for i, col in enumerate(X.T, start=5):
```

Comment:
What exactly is this doing?
It partially overwrites the regression table with blobs to avoid having two tables for the unit tests?

Reply:
Not partially, the first table has 5 features, we start at 5 here. I create a single table that as features and targets both suitable for regression and classification (like our real data)

Reply:
Thx got confused with the transposition